### PR TITLE
Feat enum refine

### LINKED ISSUE #510: Creating Enum for Estimator Mapper
**Is your feature request related to a problem? Please describe.**
e.g. create an Enum with ['stan-map] matching the class of stan map estimator

### LINKED ISSUE #511: Create Enum for Training and Prediction meta
**Is your feature request related to a problem? Please describe.**
A better way to organize the labels we used for training and prediction meta

**Describe the solution you'd like**
e.g. in the mapper, we can do num_of_obs: 'num_of_observations' etc.

### PR DESCRIPTION
**Pull request by edwinnglabs**
## Description

Please include a summary of the change and which issue is fixed.

Fixes #510 Fixes #511

### CODE REVIEW

File: orbit/diagnostics/backtest.py [hunk 1]
```diff
@@ -105,9 +107,9 @@ def _set_split_scheme(self):
             split_scheme[i] = {}
             train_start_idx = train_end_idx - self.min_train_len + 1 \
                 if self.window_type == 'rolling' else 0
```
Comment (FreeRealEstate221) [inline orbit/diagnostics/backtest.py:109]: Maybe add a rolling window type Enum here?

File: orbit/diagnostics/backtest.py [hunk 2]
```diff
@@ -274,29 +282,36 @@ def fit_predict(self):
...
+            self._train_prediction = np.concatenate(
+                (self._train_prediction, train_predictions[BacktestFitKeys.PREDICTED.value].to_numpy()))
 
             # set df attribute
             # join train
-            train_dates = train_df[date_col].rename('date', axis='columns')
-            train_response = train_df[response_col].rename('actuals', axis='columns')
-            train_values = pd.concat((train_dates, train_response, train_predictions['prediction']), axis=1)
-            train_values['training_data'] = True
+            train_dates = train_df[date_col].rename(BacktestFitKeys.DATE.value, axis='columns')
```
Comment (FreeRealEstate221) [inline orbit/diagnostics/backtest.py:294]: add Enum for 'columns' here? This str appear multiple times in the script

File: orbit/diagnostics/plot.py [hunk 3]
```diff
@@ -495,7 +269,7 @@ def plot_bt_predictions(bt_pred_df, metrics=smape, split_key_list=None,
         figsize = (16, 8)
 
     metric_vals = bt_pred_df.groupby('split_key').apply(lambda x:
```
Comment (FreeRealEstate221) [inline orbit/diagnostics/plot.py:271]: Enum for 'split_key' & 'training_data' & 'prediction'

File: orbit/diagnostics/plot.py [hunk 4]
```diff
@@ -536,3 +310,133 @@ def plot_bt_predictions(bt_pred_df, metrics=smape, split_key_list=None,
...
+    if fig_paths and export_gif:
+        import imageio
+        with imageio.get_writer('{}/orbit-backtest.gif'.format(fig_dir), mode='I', **imageio_args) as writer:
+            for fig_path in fig_paths:
+                image = imageio.imread(fig_path)
+                writer.append_data(image)
+
+# TODO: update palatte
+@orbit_style_decorator
+def metric_horizon_barplot(df, model_col='model', pred_horizon_col='pred_horizon',
```
Comment (FreeRealEstate221) [inline orbit/diagnostics/plot.py:397]: Enum for 'model', 'pred_horizon' and etc

File: orbit/forecaster/forecaster.py [hunk 5]
```diff
@@ -7,6 +7,7 @@
 from ..utils.general import is_ordered_datetime
 from ..template.model_template import ModelTemplate
 from ..estimators.stan_estimator import StanEstimatorMCMC
+from ..constants.constants import TrainingMetaKeys, PredictionMetaKeys
 
 COMMON_MODEL_CALLABLES = ['get_data_input_mapper', 'get_fitter', 'get_init_values', 'get_model_name',
```
Comment (FreeRealEstate221) [inline orbit/forecaster/forecaster.py:12]: Enum for model callables

File: orbit/template/dlt.py [hunk 6]
```diff
@@ -439,17 +441,17 @@ def set_dynamic_attributes(self, df, training_meta):
         self.cauchy_sd = max(np.abs(training_meta['response'])) / 30.0
```
Comment (FreeRealEstate221) [inline orbit/template/dlt.py:441]: Enum for 'response'

File: orbit/template/ktr.py [hunk 7]
```diff
@@ -654,7 +659,7 @@ def _set_knots_scale_matrix(self, df, training_meta):
     def _generate_tp(self, training_meta, prediction_date_array):
         """Used in _generate_seas"""
         training_end = training_meta['training_end']
```
Comment (FreeRealEstate221) [inline orbit/template/ktr.py:661]: Enum for 'training_end' & 'date_array'

File: orbit/template/ktrlite.py [hunk 8]
```diff
@@ -245,7 +250,7 @@ def _set_seasonality_attributes(self):
     def _set_validate_ktr_params(self, training_meta):
         # avoid lengthy code
         response = training_meta['response']
```
Comment (FreeRealEstate221) [inline orbit/template/ktrlite.py:252]: Enum for 'response'

File: tests/orbit/diagnostics/test_backtest.py [hunk 9]
```diff
@@ -208,7 +208,7 @@ def test_backtester_with_training_data(iclaims_training_data):
     expected_test_metrics = [x.__name__ for x in backtester._default_metrics]
 
     expected_train_metrics = list(filter(
-        lambda x: backtester._get_metric_callable_signature(x) == {'actual', 'predicted'}, backtester._default_metrics)
+        lambda x: backtester._get_metric_callable_signature(x) == {'actual', 'prediction'}, backtester._default_metrics)
```
Comment (FreeRealEstate221) [inline tests/orbit/diagnostics/test_backtest.py:211]: Enum 'actual' & 'prediction'

File: tests/orbit/models/test_ktrlite.py [hunk 10]
```diff
@@ -109,7 +109,7 @@ def test_ktrlite_level_segments(make_daily_data, level_segments):
     knots_df = ktrlite.get_level_knots()
     levels_df = ktrlite.get_levels()
     assert knots_df.shape[0] in [level_segments + 1, level_segments + 2]
-    assert levels_df.shape[0] == ktrlite.get_training_meta()['num_of_observations']
+    assert levels_df.shape[0] == ktrlite.get_training_meta()['num_of_obs']
```
Comment (FreeRealEstate221) [inline tests/orbit/models/test_ktrlite.py:112]: Enum for 'num_of_obs'

**Review by FreeRealEstate221 — COMMENTED** [review]
Found some strings that could be replaced by Enum.

File: orbit/diagnostics/backtest.py (hunk 1 above)
Comment (edwinnglabs) [reply orbit/diagnostics/backtest.py:109]: Resolved

File: orbit/diagnostics/plot.py (hunk 4 above)
Comment (edwinnglabs) [reply orbit/diagnostics/plot.py:397]: i am not sure if we should do that for function signature as it goes into function hint and docstring etc.

File: orbit/forecaster/forecaster.py (hunk 5 above)
Comment (edwinnglabs) [reply orbit/forecaster/forecaster.py:12]: let's leave this to another issue

File: tests/orbit/models/test_ktrlite.py (hunk 10 above)
Comment (edwinnglabs) [reply tests/orbit/models/test_ktrlite.py:112]: let's lump all unit test stuff enum to another issue

**Review by FreeRealEstate221 — APPROVED** [review]

**Pull request merged**